Patch from A Ashley to fix segfault on media stop https://bugzilla.gnome.org/show_bug.cgi?id=765728
Thnx christophecvr for report: http://forums.openpli.org/topic/29501-gstreamer-10/page-121#entry545136

--- a/gst-libs/gst/adaptivedemux/gstadaptivedemux.c	
+++ a/gst-libs/gst/adaptivedemux/gstadaptivedemux.c	
@@ -3528,9 +3528,12 @@ gst_adaptive_demux_wait_until (GstClock * clock, GCond * cond, GMutex * mutex,
     gst_clock_id_unref (timer.clock_id);
     g_return_val_if_reached (TRUE);
   }
+  g_assert (!timer.fired);
   /* the gst_adaptive_demux_clock_callback will signal the
      cond when the clock's single shot timer fires */
   g_cond_wait (cond, mutex);
+  if (!timer.fired)
+    gst_clock_id_unschedule (timer.clock_id);
   gst_clock_id_unref (timer.clock_id);
   return !timer.fired;
 }
